### PR TITLE
Caliptra API Enhancements :Part-2

### PR DESCRIPTION
**Pull request by rusty1968**
- Add documentation showing how to use the SocManager trait.
- Update SocManager with new trait methods.

### CODE REVIEW

File: api/src/soc_mgr.rs [hunk 1]
```diff
@@ -15,8 +72,83 @@ pub trait SocManager {
 
     fn mmio_mut(&mut self) -> Self::TMmio<'_>;
 
+    // Provide a time base for mailbox status polling loop.
     fn delay(&mut self);
 
+    /// Set up valid PAUSERs for mailbox access.
+    fn setup_mailbox_users(&mut self, apb_pausers: &[u32]) {
+        for (idx, apb_pauser) in apb_pausers.iter().enumerate() {
+            self.soc_ifc()
```
[the remark this answers is not in the document]
Comment (rusty1968) [reply api/src/soc_mgr.rs:91]: sure

File: api/src/soc_mgr.rs [hunk 2]
```diff
@@ -60,4 +203,155 @@ pub trait SocManager {
...
+        if data.len() < R::Resp::MIN_SIZE || data.len() > mem::size_of::<R::Resp>() {
+            return Err(CaliptraApiError::MailboxUnexpectedResponseLen {
+                expected_min: R::Resp::MIN_SIZE as u32,
+                expected_max: mem::size_of::<R::Resp>() as u32,
+                actual: data.len() as u32,
+            });
+        }
+
+        let mut response = R::Resp::new_zeroed();
+        response.as_bytes_mut()[..data.len()].copy_from_slice(data);
```
Comment (zhalvorsen) [inline api/src/soc_mgr.rs:328]: Is it possible to also use `read_from_prefix` here?
```suggestion
        let response = R::Resp::read_from_prefix(data).unwrap();
```

Comment (rusty1968) [reply api/src/soc_mgr.rs:328]: This actually causes panic when the response is variable size. I changed HwModel to delegate the mailbox_execute_request to SocMgr and it panicked in this line when processing caliptra_api::mailbox::GetLdevCertResp.

**Review by zhalvorsen — APPROVED** [review]

**Review by swenson — APPROVED** [review]
LGTM

**Review by korran — APPROVED** [review]

**Pull request merged**